src/seedu/addressbook/data/person/Attendance.java

```diff
@@ -11,39 +11,64 @@
 
 public class Attendance implements Printable {
 
-    /** Represents a map that links dates(kay) to attendance(value)*/
-    private Map<String, Boolean> attendanceMap = new HashMap<>();
+    /** Represents a map for each person, showing which every attendance for each date*/
+    private Map<String, Boolean> attendancePersonMap = new HashMap<>();
 
-    /** Method to add attendance */
+    /** Method to add attendance*/
     public boolean addAttendance(String date, Boolean isPresent, Boolean overWrite) {
         String inputDate = date;
-        if ("0".equals(date)) { //PMD 3.3
+        if ("0".equals(date)) {
             inputDate = new SimpleDateFormat("dd-MM-yyyy").format(new Date());
         }
-        // If there is a duplicate date
-        if (attendanceMap.containsKey(inputDate) && overWrite) {
-            attendanceMap.put(inputDate, isPresent);
+        // return true if duplicate date
+        if (attendancePersonMap.containsKey(inputDate) && overWrite) {
+            attendancePersonMap.put(inputDate, isPresent);
             return true;
-        } else if (attendanceMap.containsKey(inputDate) && !overWrite) {
+        } else if (attendancePersonMap.containsKey(inputDate) && !overWrite) {
             return true;
+        } else if (!attendancePersonMap.containsKey(inputDate) && overWrite) {
+            return false;
         } else {
-            attendanceMap.put(inputDate, isPresent);
+            attendancePersonMap.put(inputDate, isPresent);
             return false;
         }
```

Comment:
Still quite potato, maybe it'll be better to use exceptions to declare an invalid operation